File: src/main/java/com/drunkensoftball/api/roster/service/RosterService.java
package com.drunkensoftball.api.roster.service;

import com.drunkensoftball.api.roster.domain.RosterEntry;
import org.hibernate.validator.constraints.NotBlank;
import org.springframework.validation.annotation.Validated;

@Validated
public interface RosterService {

    RosterEntry addPlayerByUuid(@NotBlank(message = "Authentication token required") String token,
                                @NotBlank(message = "Team uuid required") String teamUuid,
                                @NotBlank(message = "Player UUID required") String playerUuid,
                                Integer battingPosition,
                                String fieldPosition);

    RosterEntry addGuestPlayer(@NotBlank(message = "Authentication token required") String token,
                               @NotBlank(message = "Team uuid required") String teamUuid,
                               @NotBlank(message = "Display name required.") String displayName,
                               Integer battingPosition,
                               String fieldPosition);

}

File: src/main/kotlin/com/drunkensoftball/api/roster/service/RosterService.kt
package com.drunkensoftball.api.roster.service


import com.drunkensoftball.api.auth.domain.DSAuthentication
import com.drunkensoftball.api.roster.domain.RosterEntry

interface RosterService {

    fun addPlayerByUuid(authentication: DSAuthentication,
                        teamUuid: String,
                        playerUuid: String,
                        battingPosition: Int?,
                        fieldPosition: String?): RosterEntry

    fun addGuestPlayer(authentication: DSAuthentication,
                       teamUuid: String,
                       firstName: String,
                       lastName: String?,
                       battingPosition: Int?,
                       fieldPositionString: String?): RosterEntry

}
